### DISCUSSION TITLE
Cannot load any SVG file created with LibreOffice Draw

### DISCUSSION THREAD

**Discussion by troelsstru** [senshu/Sozi#546] [Q&A]
Hi, I installed Sozi on my new laptop, but it cannot load any svg-file :-(

Can anyone help? I am NO EXPERT (just recently learned how to use the terminal), so I'm in need of simple explanations :-)

Thanks so much! Sozi is a great program! 

Here is more informaion:

I use
Ubuntu 20.04.2 LTS
Sozi release 21.08.02 

When I choose an svg-file, it doesn't give me any error message, it just doesn't load.
I have tried both by clicking "open an svg file" in the desktop version of sozi, and from a terminal: "sozi test.svg". The terminal code opens the program, but not the file.

When I press F12, here is the informaion:

__dirname=/usr/lib/sozi/resources/app/src/js	/usr/lib/sozi/resources/app/src/js/Storage.js:121 
Using handler: Impress	/usr/lib/sozi/resources/app/src/js/svg/SVGDocumentWrapper.js:182 
Uncaught (in promise) TypeError: Cannot read property 'parentNode' of null

 at Function.transform (/usr/lib/sozi/resources/app/src/js/svg/ImpressHandler.js:32)
    at Function.fromString (/usr/lib/sozi/resources/app/src/js/svg/SVGDocumentWrapper.js:191)
    at Storage.loadSVGData (/usr/lib/sozi/resources/app/src/js/Storage.js:211)
    at Storage.setSVGFile (/usr/lib/sozi/resources/app/src/js/Storage.js:195)

After that, the message "DevTools failed to load SourceMap: Could not load content for file:///usr/lib/sozi/resources/app/node_modules/pdf-lib/cjs/index.js.map: ", repeated more than 100 times with different filenames. I'll be happy to post all these lines, if they are any help.

**Comment by aumouvantsillage** [comment]
> Using handler: Impress

Can you confirm that your SVG document has been exported from Impress?

It would be great if you could send me your SVG document, to help me investigate the problem.

**Reply by troelsstru** [reply]
Thanks for your reply! I tried with several svg-files, and yes, all of them are from Impress.
Github doensn't support svg as attached file, so I had to attach it as a zip file.
[Sozi.zip](https://github.com/sozi-projects/Sozi/files/6940989/Sozi.zip)

**Reply by aumouvantsillage** [reply]
It looks like your document was not created with LibreOffice Impress but with LibreOffice Draw.
I see that the exported SVG is very similar to what Impress produces,  but there are differences that break the import in Sozi.

A fix will be available in the next release.

Generally, I do not recommend using Impress or Draw to make presentations with Sozi. While it should work in simple cases, some features of Sozi (layers, automatic alignment, transition paths) will not be available due to missing information in the SVG exported by LibreOffice. I suggest to use [Inkscape](https://inkscape.org/) if you can.

**Reply by troelsstru** [reply]
Ok, thanks a lot for your help. I will use Inkscape. Also, it turns out that I can take my svg files created in Draw, import them to Inkscape, save it as a new svg file, and then it works in Sozi.
